lib/exsom.ex

```diff
@@ -12,8 +12,23 @@ defmodule Exsom do
   The most common functions get exposed as defaults in the parent module.
   """
 
-  @type xml :: [any] | [char] | binary
-  @type raw_xml :: [integer] | binary
+  @type binary_xml    :: binary       # XML document in binary form
+  @type charlist_xml  :: charlist     # XML document in charlist form
+  @type decoded_xml   :: [integer]    # XML document represented by unicode code points
+
+  @type xml           :: decoded_xml | charlist_xml | binary_xml
+  @type raw_xml       :: decoded_xml | binary_xml
+
+  @type binary_xsd    :: binary       # XSD schema in binary form
+  @type charlist_xsd  :: charlist     # XSD schema in charlist form
+  @type decoded_xsd   :: [integer]    # XSD schema represented by unicode code points
+
+  @type xsd           :: decoded_xsd | charlist_xsd | binary_xsd
+  @type raw_xsd       :: decoded_xsd | binary_xsd
+
+  @typedoc "Internal representation of a XSD document"
+  @type model         :: any
+
```

Comment:
Maybe just a note about the use case would clarify for readers of the docs down the road (ie: us in 3 weeks 😉). Even just

```elixir
@type decoded_xml   :: [integer]    # XML document represented by Unicode code points (ie: a list of integers)
```